### PR TITLE
give a default value to T and convert models from v1.2

### PR DESCRIPTION
**Pull request by njzjz**
See also https://www.tensorflow.org/guide/create_op#backwards_compatibility.

Implement #546.

### CODE REVIEW

**Review by amcadmus — APPROVED** [review]

**Comment by amcadmus** [comment]
How about adding a section in the [getting started ](https://github.com/deepmodeling/deepmd-kit/blob/devel/doc/getting-started.md) to introduce the users the conversions of model compatibility?

**Comment by njzjz** [comment]
I add a table to troublesome and I am not sure if it should be moved to the getting started.

**Review by tuoping — CHANGES_REQUESTED** [review]
How about add a link to doc/troubleshooting/model-compatibility in getting-started?

File: doc/troubleshooting/model-compatability.md [hunk 1]
```diff
@@ -1,5 +1,21 @@
...
+
+One can execuate `dp convert-from` to convert an old model to a new one.
+
+|   | Model v0.12 | Model v1.0 | Model v1.1 | Model v1.2 | Model v1.3 | Model v2.0 |
+|:-:|:-----------:|:----------:|:----------:|:----------:|:----------:|:----------:|
+|DeePMD-kit v0.12 | 😄 | 😢 | 😢 | 😢 | 😢 | 😢 |
+|DeePMD-kit v1.0  | 😢 | 😄 | 😢 | 😢 | 😢 | 😢 |
+|DeePMD-kit v1.1  | 😢 | 😢 | 😄 | 😢 | 😢 | 😢 |
+|DeePMD-kit v1.2  | 😢 | 😢 | 😢 | 😄 | 😢 | 😢 |
+|DeePMD-kit v1.3  | 😢 | 😢 | 😢 | 😊 | 😄 | 😢 |
```
Comment (amcadmus) [inline doc/troubleshooting/model-compatability.md:15]: Do we have tools to convert v1.3 model to v1.2 compatibility?

**Review by denghuilu — APPROVED** [review]
works well in my local workstation

File: doc/troubleshooting/model-compatability.md (hunk 1 above)
Comment (njzjz) [reply doc/troubleshooting/model-compatability.md:15]: No.

Comment (amcadmus) [reply doc/troubleshooting/model-compatability.md:15]: So why there is a smile? Is it a typo?

Comment (njzjz) [reply doc/troubleshooting/model-compatability.md:15]: The smile is where model version is v1.2 and deepmd-kit version is 1.3, not converse.

Comment (amcadmus) [reply doc/troubleshooting/model-compatability.md:15]: Can we use a v1.2 model in deepmd-kit v1.3?

Comment (amcadmus) [reply doc/troubleshooting/model-compatability.md:15]: So you mean we can convert a model v1.2 to v1.3 by using deepmd-kit v1.3.   I do not see this subcommand in v1.3

Comment (njzjz) [reply doc/troubleshooting/model-compatability.md:15]: We can convert a model v1.2 to v1.3 by using deepmd-kit v1.2. So this should be a 😢?

Comment (amcadmus) [reply doc/troubleshooting/model-compatability.md:15]: Yes, the manner of conversion is somewhat different... In deepmd-kit v1.2 and 1.3, it is `convert-to` (pending PR #631 and #633), but in v2.0 it is `convert-from`.

I would proposed to only keep the doc for v2.0, and write separate docs for v1.2 and 1.3, what do you think?

Comment (njzjz) [reply doc/troubleshooting/model-compatability.md:15]: Good idea.

**Review by amcadmus — APPROVED** [review]

**Pull request merged**